.github/workflows/inspektor-gadget.yml

```diff
@@ -258,7 +245,7 @@ jobs:
         # Answer: push to runner registry first, if integration tests are OK
         # push to final registry
```

Comment:
I tried on my fork to push a tag https://github.com/mauriciovasquezbernal/inspektor-gadget/tree/v0.2.0.test4, the container image for this tag was pushed to https://github.com/mauriciovasquezbernal/inspektor-gadget/pkgs/container/inspektor-gadget/18476048?tag=v0.2.0.test4 before running the instegration tests, that in this case, they failed https://github.com/mauriciovasquezbernal/inspektor-gadget/runs/5854720480?check_suite_focus=true. 

What I mean it, even for the main branch of for tags, we should run integration tests before pushing the image to the final repo.

Reply:
Oh! This case is expected due to the code.
I will need to think about it but I would like to avoid to push images in several places...

Reply:
Why is this complicated. This PR already has two different repos: `inspektor-gadget-ci` and `inspektor-gadget`, we can use the former for all tests (integration and so on), and only push to the latter when pushing a release (or to the main branch).

Reply:
> Why is this complicated.

Complicated is not forcefully the good word, complex would be better.
With your solution, I will need to push twice and I am not happy pushing twice because well "twice".
The best solution would be to build the image locally, store it in cache or artifact and give it to integration tests step but this will not work when we will add ARO.

Nonetheless, I think I do not have other choices than to push twice, this is a solution but I do not like it.

Reply:
> The best solution would be to build the image locally, store it in cache or artifact and give it to integration tests step

I don't see how it's less complex than pushing to the "ci" registry :smile:

Reply:
I agree, than using artifact is not simple in terms of code, but at least we do not interact with ouside registry.
I normally a pushed a change which should address the TODO but I would like to test it thoroughly.